### PR TITLE
[CI] Limit number of threads in all jobs

### PR DESCRIPTION
**Pull request by tqchen**
To reduce pressure on a crowded CI setting.

### CODE REVIEW

**Comment by tqchen** [comment]
cc @yzhliu @zhiics @icemelon9 @tmoreau89

File: tests/scripts/task_python_vta_fsim.sh [hunk 1]
```diff
@@ -20,6 +20,10 @@ set -e
 set -u
 
 source tests/scripts/setup-pytest-env.sh
+# to avoid CI thread throttling.
+export TVM_BIND_THREADS=0
+export OMP_NUM_THREADS=1
+
```
Comment (u99127) [inline tests/scripts/task_python_vta_fsim.sh:26]: Why not move this into setup-pytest-env.sh and maybe rename it to setup-test-env.sh if you think it's too pytest centric ?

Comment (u99127) [reply tests/scripts/task_python_vta_fsim.sh:26]: Thanks for the reply. Agreed on the rust and cpp environments which is why I suggested setup-test-env.sh :). On the second point I would have thought the lower level driver script be able to override the environment variable for the test ? 

I'd like to understand more about what you mean by directly limiting the resources of the container. 

regards
Ramana

Comment (tqchen) [reply tests/scripts/task_python_vta_fsim.sh:26]: I meant https://docs.docker.com/config/containers/resource_constraints/, to directly restrict the CPUs that a container can access physically. although we might need some experimentation to see which option is the best, and how are they presented to the runtime.

**Comment by tqchen** [comment]
Merging for now as the current loc changes is small, we can do more delibration about option consolidation as follow ups, Thanks @u99127 @zhiics !

**Pull request merged**